src/main/java/org/jenkinsci/lib/configprovider/ConfigProvider.java

```diff
@@ -128,4 +129,46 @@ public boolean supportsFolder() {
         return true;
     }
 
+
+    /**
+     * @deprecated Use <tt>GlobalConfigFiles.get().getConfigs()</tt> instead.
+     */
+    @Deprecated
+    public Collection<Config> getAllConfigs() {
```

Comment:
The correct way to access configuration files is via `org.jenkinsci.plugins.configfiles.ConfigFiles` and not via `GlobalConfigFiles` - `GlobalConfigFiles` should never be accessed directly.

Readding these methods to this class for sure cause wrong usage.

So I'm really puzzled about this change...